frame/contracts/src/storage/meter.rs

```diff
@@ -224,6 +224,27 @@ impl Diff {
 	}
 }
 
+/// The beneficiary of a contract termination.
+#[derive(RuntimeDebugNoBound, Clone, PartialEq, Eq)]
+pub struct Beneficiary<T: Config>(AccountIdOf<T>);
+
+impl<T: Config> Deref for Beneficiary<T> {
+	type Target = AccountIdOf<T>;
+
+	fn deref(&self) -> &Self::Target {
+		&self.0
+	}
+}
+
+/// The status of a contract.
+///
+/// In case of termination the beneficiary is indicated.
+#[derive(RuntimeDebugNoBound, Clone, PartialEq, Eq)]
+pub enum ContractStatus<T: Config> {
+	Alive,
+	Terminated(Beneficiary<T>),
```

Comment:
This is what I meant by struct like variant. No need for an alias or newtype just to give something a name.

```suggestion
	Terminated { beneficiary: AccountIdOf<T> },
```